File: stitch_smartware_official_website/fix-regex.js
const fs = require('fs');
const path = require('path');

const files = fs.readdirSync(__dirname)
    .filter(f => f.startsWith('product-') && f.endsWith('.html'));

let fixed = 0;
for (const file of files) {
    const filepath = path.join(__dirname, file);
    let content = fs.readFileSync(filepath, 'utf8');

    // The broken pattern: var match = src.match(/images\([^.-]+)/);
    // Use string literal that preserves backslashes
    const searchStr = 'var match = src.match(/images' + String.raw`\([^.-]+\)` + '/);';
    const replaceStr = 'var match = src.match(/images\\/([^.-]+)/);';

    if (content.includes(searchStr)) {
        content = content.replace(searchStr, replaceStr);
        fs.writeFileSync(filepath, content, 'utf8');
        console.log('FIXED: ' + file);
        fixed++;
    }
}

console.log('Done. Fixed ' + fixed + ' files.');

File: stitch_smartware_official_website/fix-regex.py
import os
import glob

website_dir = "d:/CC/smartware/Version2/stitch_smartware_official_website"
files = glob.glob(os.path.join(website_dir, "product-*.html"))

fixed = 0
for filepath in files:
    with open(filepath, 'rb') as f:
        content = f.read()

    # Fix: /images\(... -> /images/...
    old = b'images\\\\([^.-]+)'
    new = b'images\\/([^.-]+)'

    if old in content:
        content = content.replace(old, new)
        with open(filepath, 'wb') as f:
            f.write(content)
        fixed += 1

print(f"Fixed {fixed} files")
